app/controllers/AuthenticationController.scala

```diff
@@ -234,39 +234,61 @@ class AuthenticationController @Inject()(
    */
 
   def accessibleBySwitching(datasetId: Option[String],
+                            datasetDirectoryName: Option[String],
+                            organizationId: Option[String],
                             annotationId: Option[String],
                             workflowHash: Option[String]): Action[AnyContent] = sil.SecuredAction.async {
     implicit request =>
       for {
         datasetIdValidated <- Fox.runOptional(datasetId)(ObjectId.fromString(_))
+        _ <- Fox
+          .runIf(datasetDirectoryName.isDefined)(bool2Fox(organizationId.isDefined)) ?~> "Organization id must be defined if dataset directory name is defined."
```

Comment:
```suggestion
          bool2Fox(datasetDirectoryName.isDefined == organizationId.isDefined)) ?~> "Organization id must be defined if dataset directory name is defined."
```
that’s equivalent, right? Might be more easy to read